### PR TITLE
new: Add support for loading JSON OpenAPI spec files

### PR DESCRIPTION
**Pull request by lgarber-akamai**
## 📝 Description

This pull request adds support for loading the Linode OpenAPI spec as either a JSON or YAML file, which will prepare the CLI for the move away from the legacy spec publication system.

**NOTE: This pull request previously included the removal of the linode-api-docs latest version resolution, but will be moved to a separate PR due to the delayed move to the akamai-apis repository.**

## ✔️ How to Test

The following test steps assume you have pulled down this PR locally. 

Installing with the TechDocs OpenAPI spec (JSON): 

```
make SPEC="https://gist.githubusercontent.com/lgarber-akamai/06b1e4c6e3578b936e293461c03e7feb/raw/e6917ef9f54add09d2716c93e332dc11ed330932/openapi.json" install
```

Installing with the Legacy OpenAPI spec (YAML):

```
make install
```

### Integration Testing

```bash
make testint
```

### Unit Testing

```bash
make testunit
```

### Manual Testing

Manual testing can be done by simply running `linode-cli` commands after running one of the above installation commands.

### CODE REVIEW

File: linodecli/cli.py
Comment (lgarber-akamai) [inline linodecli/cli.py:245]: Given the new OpenAPI spec is so large, I used a ContextManager here to avoid loading the spec into memory more times than we need to.

File: tests/unit/conftest.py
Comment (lgarber-akamai) [inline tests/unit/conftest.py:32]: This is necessary so we can manually load raw fixture contents in unit tests

File: tests/unit/test_arg_helpers.py
Comment (lgarber-akamai) [inline tests/unit/test_arg_helpers.py:154]: This is no longer relevant

Comment (lgarber-akamai) [inline tests/unit/test_arg_helpers.py:154]: I decided to remove this error handler/test because the detailed error gives more valuable information (especially when it comes to cert issues):

```
Failed to load spec: HTTPSConnectionPool(host='fake.linode.com', port=443): Max retries exceeded with url: / (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x10f38a5d0>: Failed to resolve 'fake.linode.com' ([Errno 8] nodename nor servname provided, or not known)"))
```

File: Makefile
Comment (lgarber-akamai) [inline Makefile:5]: Just leaving myself a reminder to replace this before merging

File: linodecli/cli.py
Comment (zliang-akamai) [inline linodecli/cli.py:244]: Very cool usage of context manager!

**Review by zliang-akamai — APPROVED** [review]
Works well on my side, nice work!

**Review by yec-akamai — APPROVED** [review]
Tested locally and works great, nice work!

**Review by yec-akamai — APPROVED** [review]
Great work!

**Review by zliang-akamai — APPROVED** [review]

**Review by zliang-akamai — APPROVED** [review]

**Review by yec-akamai — APPROVED** [review]
LGTM! Just FYI when I run the integration tests with different specs, they throw some different errors. Probably because the spec versions are different and have different change.

**Pull request merged**
